util.py

```diff
@@ -419,6 +421,93 @@ def winedll_override(dll: str, dtype: Literal['n', 'b', 'n,b', 'b,n', '']) -> No
     )
 
 
+def patch_libcuda() -> bool:
+    """Patches libcuda to work around games that crash with DLSS due to wonky memory allocation.
+
+    The patched library is overwritten at every launch and is placed in .cache
+
+    Returns true if the library was patched correctly. Otherwise returns false
+    """
+    cache_dir = os.path.expanduser('~/.cache/protonfixes')
+    os.makedirs(cache_dir, exist_ok=True)
+
+    try:
+        library_base = ctypes.util.find_library('cuda')
+        if not library_base:
+            log.warn('libcuda.so not found on the system.')
+            return False
+
+        library_paths = [
+            '/lib64', '/usr/lib64', '/usr/local/lib64',
+            '/lib', '/usr/lib', '/usr/local/lib'
+        ]
```

Comment:
We actually can simplify this a little bit, depending on your goal.

Proton is intended to be use within the Steam Runtime, and the base platform is Debian-based where it's shared library paths can be known ahead of time -- `/usr/lib/x86_64-linux-gnu` and `/usr/lib/i386-linux-gnu`. This was true in the past (steamrt2 AKA soldier) and will be in the future with steamrt4 and steamrt5. 

But you can't find this file there, since it's part of the graphics driver stack. The graphics stack that's mounted by pressure-vessel will be in `/usr/lib/pressure-vessel/overrides/` so you will need to find the file there. The `proton` file makes this simple by setting  `LD_LIBRARY_PATH=LD_LIBRARY_PATH=/usr/lib/pressure-vessel/overrides/lib/x86_64-linux-gnu/aliases:/usr/lib/pressure-vessel/overrides/lib/i386-linux-gnu/aliases`. Just be sure to move up from the `aliases` directory.

Reply:
But if your goal is to also support Proton not being run inside its runtime, parse the output `ldconfig -p` or `ldconfig -XNv` for the path candidates. The Python standard library actually does the former and this approach would be a more general solution than the above.

Reply:
Also, if you're interested to see the files in the Steam Runtime container for debugging purposes, see:

https://gitlab.steamos.cloud/steamrt/steam-runtime-tools/-/blob/main/docs/slr-for-game-developers.md?ref_type=heads#running-in-an-interactive-shell

Reply:
I see.. I'll have to rework this a bit. I was initially intending to avoid anything non-python for portability's sake, but calling ldconfig would work too as cutils does that as well like you said. My initial attempt was to avoid anything outside of python as a bit of a challenge, but parsing the output of ldconfig simplifies this a fair bit. 

My mind is still stuck in the days of pre-pressure vessel proton.. whoops. Tthe Linux runtime never occurred to me, I'll poke around in a shell to check on locations of the library.

Reply:
I used ldconfig -p's output to find the correct library. This is nice as ldconfig will also tell us which arch the library is (hence the check for the x86-64 string) which allows me to remove the need for checking if the ELF file is indeed a 64bit binary or not.

https://github.com/Open-Wine-Components/umu-protonfixes/pull/136/commits/8a27ea21d16aac2ba4cf2734450edd07b70c6b47 fixes this, and should pull in the correct paths as it searches overrides as well since `ld.so.cache` should be updated with libraries within them.